### DISCUSSION TITLE
Rails application layout is not being used, don't see the Vite (bolt) Ruby console message

### DISCUSSION THREAD

**Discussion by wdiechmann** [ElMassimo/vite_ruby#289] [Q&A] [answered]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [x] I have tried upgrading by running `bundle update vite_ruby`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

Doing a clean install and following the 'getting started' in the Vite documentation should leave me with a console message 

### Reproduction 🐞

```
rails new vite
cd vite
bundle add vite_rails
bundle exec vite install
vi package.json ( getting rid of license missing error)
rm package-lock.json
yarn
bin/vite dev
```

<details>
<summary>Vite Ruby Info</summary>

_Run `bin/rake vite:info` and provide the output:_

```
bin/vite present?: true
vite_ruby: 3.2.7
vite_rails: 3.0.12
rails: 7.0.4
node: v18.2.0
npm: 8.9.0
yarn: 1.22.19
pnpm: 
ruby: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]

installed packages:
vite@ /Users/walther/Deling/Walther/src/testing/vite
├─┬ vite-plugin-ruby@3.1.2
│ └── vite@3.1.8 deduped
└── vite@3.1.8
```
</details>

### Logs 📜

_If not providing a reproduction:_

<details>
<summary>Output</summary>

Hitting localhost:3000 does show the expected Rails default page -- while running `rails s` and `bin/vite dev` in two consoles, but the console.log message is nowhere to be seen -- all I get is provided in the attached screenshot from the devtools window

```
Building with Vite ⚡️
vite v3.1.8 building for production...
transforming...
✓ 1 modules transformed.
rendering chunks...
../../public/vite/manifest-assets.json             0.00 KiB
../../public/vite/manifest.json                    0.14 KiB
../../public/vite/assets/application.ff486227.js   0.13 KiB / gzip: 0.13 KiB
../../public/vite/assets/application.ff486227.js.map 1.09 KiB
Build with Vite complete: /Users/walther/Deling/Walther/src/testing/vite/public/vite
2022-10-21T09:21:47.739Z vite-plugin-ruby:config {
  base: '/vite/',
  build: {
    emptyOutDir: false,
    sourcemap: true,
    assetsDir: 'assets',
    manifest: true,
    outDir: '../../public/vite',
    rollupOptions: { input: [Object], output: [Object] }
  },
  envDir: '/Users/walther/Deling/Walther/src/testing/vite',
  root: '/Users/walther/Deling/Walther/src/testing/vite/app/javascript',
  server: {
    fs: { allow: [Array], strict: true },
    host: 'localhost',
    https: false,
    port: 3036,
    strictPort: true,
    hmr: { clientPort: 3036 }
  },
  entrypoints: {
    'entrypoints/application.js': '/Users/walther/Deling/Walther/src/testing/vite/app/javascript/entrypoints/application.js'
  }
}
2022-10-21T09:21:47.797Z vite-plugin-ruby:assets-manifest { manifest: Map(0) {} }

```
</details>

### Screenshots 📷

_Provide console or browser screenshots of the problem_.

![Skærmbillede 2022-10-21 kl  11 26 38](https://user-images.githubusercontent.com/193245/197162798-2bc8b2e2-e155-4238-b7ef-e975dfa883c4.png)

**Comment by ElMassimo** [comment]
Please follow the steps in the [__Setup__](https://vite-ruby.netlify.app/guide/#setup-%F0%9F%93%A6) section.

In particular, verify that [tag helpers](https://vite-ruby.netlify.app/guide/development.html#tag-helpers-%F0%9F%8F%B7) are being injected in the layout for the view that you are rendering.

**Comment by wdiechmann** [comment]
ehh @ElMassimo - what part of the setup did I miss?

There's hardly any need for me to provide a repo as it's 8 cmds off the command line -- anyone should be able to confirm it -- but nevermind, I just wanted to draw your attention to the issue (which I take it, given that you moved it to 'discussions',  is not an issue, so: we're good)

**Reply by wdiechmann** [reply]
but - just so that we are clear - I see no js loaded which puzzles me as @quaasi asked [this question](https://github.com/ElMassimo/vite_ruby/discussions/233) back in July stating that he did infact see the console message -

![Skærmbillede 2022-10-21 kl  15 40 29](https://user-images.githubusercontent.com/193245/197210656-eeaf6608-206e-4919-a583-b7aae63efb97.png)

**Reply by ElMassimo** [reply]
It seems that you haven't added any routes, the HTML you are seeing is the Ruby on Rails default template.

As a result, it's not using the layout on the screenshot on the left.

Try adding your first controller/view, and it should use that layout.

**Comment by ElMassimo** [comment, marked as answer]
Based on [the provided screenshot](https://user-images.githubusercontent.com/1158253/197233882-e4e51fff-9677-4623-bea2-9e64db066251.png) it seems that you haven't added any routes, the HTML you are seeing is the Ruby on Rails default template.

As a result, it's not using the layout on the screenshot on the left.

Try adding your first controller/view, and it should use that layout.

**Comment by wdiechmann** [comment]
TIL that Rails layouts are not used unless you offer at least one route - I have been hacking on Rails since 2008 but I've never hit this 

Thank you for setting time aside for an old man!

**Reply by wdiechmann** [reply]
oh - and the issue title edit - classy way of calling n00b 🤣

**Reply by ElMassimo** [reply]
Haha, not like that. I always edit Discussion titles so that they are easier to find in Google 😃 

Glad that it works now!

**Comment by wdiechmann** [comment]
Anyways I’m going to work my way through vite - it looks promising even if importmaps are on my Xmas wishlist too 😊

Great work! 

Med venlig hilsen
Walther
